store/tikv/lock_resolver.go

```diff
@@ -309,21 +310,24 @@ func (lr *LockResolver) ResolveLocks(bo *Backoffer, callerStartTS uint64, locks
 			if err != nil {
 				msBeforeTxnExpired.update(0)
 				err = errors.Trace(err)
-				return msBeforeTxnExpired.value(), err
+				return msBeforeTxnExpired.value(), nil, err
 			}
 		} else {
 			tikvLockResolverCountWithNotExpired.Inc()
 			// If the lock is valid, the txn may be a pessimistic transaction.
 			// Update the txn expire time.
 			msBeforeLockExpired := lr.store.GetOracle().UntilExpired(l.TxnID, status.ttl)
 			msBeforeTxnExpired.update(msBeforeLockExpired)
+			if callerStartTS > 0 {
+				pushed = append(pushed, l.TxnID)
+			}
```

Comment:
do we have rolling upgrade problem here?

old TiDB B write some locks with `lock.minCommitTS == 0`, new TiDB A read same key with `callerStartTS > 0`...

so those lock returned but not be pushed?  @tiancaiamao  @coocood 

https://github.com/pingcap/tidb/blob/8dc46b20cc7696ad8728b18f25c155b0c067cb98/store/mockstore/mocktikv/mvcc_leveldb.go#L1074

Reply:
Good catch, we should try to push the transaction even if minCommitTS == 0

Reply:
@tiancaiamao @MyonKeminta

Reply:
Addressed.
We don't push `minCommitTS == 0` and this information is returned, `ResolveLocks` will not set the `pushed` result in this case (thus the caller has to wait).